scalyr_agent/tests/monitors_manager_test.py

```diff
@@ -101,80 +91,37 @@ def test_user_agent_polling(self):
+            null_logger=True,
+            fake_clock=fake_clock,
+            set_daemon=True,
+        )
         self.assertEquals(test_manager._user_agent_refresh_interval, 30)  # ensure config setting works
         self.assertEquals(len(test_manager.monitors), 3)
         patched_monitor_0 = test_manager.monitors[0]
         patched_monitor_1 = test_manager.monitors[1]
         unpatched_monitor = test_manager.monitors[2]
 
-        # Fake the clock to fast-forward MonitorsManager sleep loop
-        fake_clock = scalyr_util.FakeClock()
-        for obj in [test_manager, patched_monitor_0, unpatched_monitor]:
-            obj._run_state = scalyr_util.RunState(fake_clock=fake_clock)
-
-        class FragmentPolls(object):
-            def __init__(self):
-                self.__polls = 0
-                self.__condition = threading.Condition()
-
-            def count(self):
-                self.__condition.acquire()
-                try:
-                    return self.__polls
-                finally:
-                    self.__condition.release()
-
-            def increment(self):
-                self.__condition.acquire()
-                try:
-                    self.__polls += 1
-                    self.__condition.notifyAll()
-                finally:
-                    self.__condition.release()
-
-            def wait_for_increment(self, old_count, timeout=None):
-                remaining = timeout
-                self.__condition.acquire()
-                try:
-                    while self.__polls == old_count and remaining > 0:
-                        t1 = time.time()
-                        self.__condition.wait(remaining)
-                        remaining -= time.time() - t1
-                finally:
-                    self.__condition.release()
-
-            def poll_until_target(self, target_polls, maxwait):
-                """Blocks until n fragment polls are simulated by the MonitorsManager, or an absolute cutoff time.
-                whichever comes first.
-
-                @param target_polls: Number of polls to reach
-                @param maxwait: Time (seconds) to wait for target to be reached
-                @return: True if number of polls reaches target_polls, else False
-                """
-                deadline = time.time() + maxwait
-                while self.count() < target_polls and time.time() < deadline:
-                    fake_clock.block_until_n_waiting_threads(1)  # wait for monitor thread loop to complete
-                    old_count = self.count()
-                    fake_clock.advance_time(increment_by=poll_interval)
-                    self.wait_for_increment(old_count, timeout=deadline-time.time())
-
-                return self.count() == target_polls
-
-        fragment_polls = FragmentPolls()
+        # The MonitorsManager + 3 monitor threads will wait on the fake clock
+        fragment_polls = FakeClockCounter(fake_clock, num_waiters=4)
```

Comment:
The related Fake Clock and Counter have been encapsulated into `FakeClockCounter` for reuse across multiple tests.